lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1159,6 +1168,84 @@ def _revoke_certificates_for_which_no_csr_exists(self, relation_id: int) -> None
                 )
                 self.remove_certificate(certificate=certificate["certificate"])
 
+    def get_requirer_units_csrs_with_no_certs(
+        self,
+    ) -> List[Dict[str, Union[int, str, List[Dict[str, str]]]]]:
+        """Filters the requirer's units csrs.
+
+        Keeps the ones for which no certificate was provided.
+
+        Returns:
+            list: List of dictonries that contain the unit's csrs
+            that don't have a certificate issued.
+        """
+        all_unit_csr_mappings = copy.deepcopy(self.get_requirer_csrs_by_unit())
+        for unit_csrs in all_unit_csr_mappings:
+            base64_keys_list = list(
+                map(
+                    lambda item: base64.b64encode(item.encode()).decode(),
+                    self.get_issued_certificates(unit_csrs["relation_id"])[  # type: ignore[arg-type]
+                        unit_csrs["application_name"]  # type: ignore[index]
+                    ].keys(),
+                )
+            )
+            for csr in unit_csrs["unit_csrs"]:  # type: ignore[union-attr]
+                if csr["certificate_signing_request"] in base64_keys_list:  # type: ignore[index]
+                    unit_csrs["unit_csrs"].remove(csr)  # type: ignore[union-attr, arg-type]
+            if len(unit_csrs["unit_csrs"]) == 0:  # type: ignore[arg-type]
+                all_unit_csr_mappings.remove(unit_csrs)
+        return all_unit_csr_mappings
+
+    def get_requirer_csrs_by_unit(
```

Comment:
The method name makes us think that we need to provide a unit name or id. Reality is that we simply return all CSR's.

I think `get_requirer_csrs` makes more sense. 

I see we already have a `_requirer_csrs` property. It feels like we probably don't need these 2 different methods and could consolidate them into 1.

Reply:
`_requirer_csrs` is part of the requirer's interface, it uses `self.model.unit` to get the csrs of the requirer unit running the code.
It's a bit different from what we are doing, and extracting the common logic to a global functions wouldn't save much code duplication in my opinion.
We might want to rename this however.

Reply:
`_unit_csrs` might be a better name for this property.